### DISCUSSION TITLE
[Question] How do I bind a command to a button within a listview?

### DISCUSSION THREAD

**Discussion by waylaa** [reactiveui/ReactiveUI#3049] [Q&A]
Hello.

I would like to know if it's a possible to bind a command to a button that is generated in cell template inside of a listview. 
Here's the code, the command thing i am talking about is in the last cell template:

        <ListView Height="auto"
                        Width="auto"
                        Margin="5"
                        Grid.Column="0"
                        Grid.Row="0"
                        Style="{DynamicResource MaterialDesignListView}"
                        x:Name="DetectedAdbDevices">

            <ListView.View>
                <GridView>
                    <GridViewColumn Header="Device"
                                                   Width="150">

                        <GridViewColumn.CellTemplate>
                            <DataTemplate>
                                <TextBlock Margin="5"
                                                  HorizontalAlignment="Center"
                                                  Text="{Binding Name}"
                                                  Foreground="{DynamicResource MaterialDesignDarkForeground}" />

                            </DataTemplate>
                        </GridViewColumn.CellTemplate>
                    </GridViewColumn>

                    <GridViewColumn Header="Model"
                                                   Width="150">

                        <GridViewColumn.CellTemplate>
                            <DataTemplate>
                                <TextBlock Margin="5"
                                                  HorizontalAlignment="Center"
                                                  Text="{Binding Model}"
                                                  Foreground="{DynamicResource MaterialDesignDarkForeground}" />

                            </DataTemplate>
                        </GridViewColumn.CellTemplate>
                    </GridViewColumn>

                    <GridViewColumn Header="State"
                                                   Width="150">
                        
                        <GridViewColumn.CellTemplate>
                            <DataTemplate>
                                <TextBlock HorizontalAlignment="Center"
                                                   Text="{Binding State}"
                                                   Foreground="{DynamicResource MaterialDesignDarkForeground}" />
                                
                            </DataTemplate>
                        </GridViewColumn.CellTemplate>
                    </GridViewColumn>

                    <GridViewColumn>
                        <GridViewColumn.CellTemplate>
                            <DataTemplate>
                                <Button HorizontalAlignment="Center"
                                              Content="{Binding Connection}" />

                            </DataTemplate>
                        </GridViewColumn.CellTemplate>
                    </GridViewColumn>
                </GridView>
            </ListView.View>
        </ListView>

**Comment by RLittlesII** [comment]
There are a few options, depending on what you are trying to accomplish and your preference for approach.  The number one question I would ask.  Do you want each element to have a `Command` that is bound to its own button, or would you prefer each item button in the list bound to the same `Command`?

1. You can use [Relative Source Binding](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/app-fundamentals/data-binding/relative-bindings) to bind your button at the View Model to the item in the list view.  You might want to consider passing the item as a `CommandParameter`
2. You can format your `ListView.View` into a control and do some [Control Template](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/app-fundamentals/templates/control-template) magic to allow you to bind.

This example does the later, and it sends you down the path of a button on each item being bound to it's own command.

[View Cell](https://github.com/RLittlesII/Reactive.Sandbox/blob/master/xamarin/forms/Removeable/Removeable/MainPage.xaml)

**Reply by waylaa** [reply]
I want each element bound to the same button because when clicked it will do the same thing regardless of the element. The thing is, i have no idea how something like this can be implemented. I can provide more code if you want, probably from the viewmodel this time.
